### PR TITLE
Upgrade the stripe-ruby-mock to 3.1.0.rc3 to support the ruby 3 upgrade

### PR DESCRIPTION
**Pull request by jiaqi-gu**
Asana Task: https://app.asana.com/0/1201719037010014/1201802388422597/f

- Remove unnecessary tests. 
- Add additional data. 
- Support updating subscription with multiple plan items.

### CODE REVIEW

File: lib/stripe_mock/data.rb [hunk 1]
```diff
@@ -351,15 +351,18 @@ def self.mock_subscription(params={})
           data: [{
             id: 'si_1AwFf62eZvKYlo2C9u6Dhf9',
             created: 1504035973,
-            metadata: {},
+            metadata: {
+              company_deleted: false
```
Comment (gz2ca) [inline lib/stripe_mock/data.rb:356]: Thoughts on adding some comments to denote what each of change in this PR is needed for?

File: lib/stripe_mock/request_handlers/validators/param_validators.rb [hunk 2]
```diff
@@ -65,71 +65,9 @@ def invalid_currency_message(my_val)
         "Invalid currency: #{my_val.downcase}. Stripe currently supports these currencies: #{SUPPORTED_CURRENCIES.join(", ")}"
       end
 
-      def validate_create_plan_params(params)
```
Comment (gz2ca) [inline lib/stripe_mock/request_handlers/validators/param_validators.rb:68]: Curious, do all the validations cause issues for us? Wonder if it's possible to identify the specific validations that cause issues and remove those ones specifically? I think it'll still be helpful to keep validations if we can to keep test behavior more similar to prod. What do you think?

Comment (jiaqi-gu) [reply lib/stripe_mock/request_handlers/validators/param_validators.rb:68]: Yes, the price, product and currency check all failed. Had to remove it entirely. 

In the current version, it doesn't have these 2 validation methods. The plan is to fix the tests or the validations (e.g. the currency validation does not take symbols or lower cases). For now, I am removing them for unblocking ruby 3 upgrade, and will add a task in the backlog to revisit it.

File: lib/stripe_mock/data.rb (hunk 1 above)
Comment (jiaqi-gu) [reply lib/stripe_mock/data.rb:356]: Added comments.

File: lib/stripe_mock/request_handlers/validators/param_validators.rb (hunk 2 above)
Comment (gz2ca) [reply lib/stripe_mock/request_handlers/validators/param_validators.rb:68]: LOL gotcha, thank you for the additional context! Follow-up task sounds good to me 😄 I think we should probably consider upgrading our Stripe API version soon too as stripe-ruby-mock is now supporting a newer API version than the one we are using...

File: lib/stripe_mock/request_handlers/helpers/subscription_helpers.rb [hunk 3]
```diff
@@ -52,6 +52,24 @@ def custom_subscription_params(plans, cus, options = {})
           params.merge!({status: 'trialing', current_period_end: end_time, trial_start: start_time, trial_end: end_time, billing_cycle_anchor: nil})
         end
 
+        # Support updating subscription with multiple plan items
```
Comment (gz2ca) [inline lib/stripe_mock/request_handlers/helpers/subscription_helpers.rb:55]: Just noting for posterity: change from https://github.com/docsend/stripe-ruby-mock/pull/1/files#diff-ce78c328fa7b9205861fad8c25d908164c4e6f7a5d1fbd2caecef970418046afR27

**Review by gz2ca — APPROVED** [review]
Woot woot looks good to me! Thoughts on keeping this PR unmerged so it's easier to see our changes on top of base?

One other tiny NIT is thoughts on updating the PR title/description to be something more descriptive? ex. maybe something related to upgrading the gem to 3.1.0.rc3 to support the ruby 3 upgrade?

**Pull request merged**